File: src/main/java/ua/lviv/iot/models/CampingProduct.java
package ua.lviv.iot.models;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
		
@Entity
@Inheritance
public class CampingProduct {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)

	private Integer id;

	private String producerName;
	private double price;
	private double weight;
	
	public CampingProduct() {}
	
	public CampingProduct(double price, double weight, String producerName) {
		this.price = price;
		this.weight = weight;
		this.producerName = producerName;
	}
	
	public double getPrice() {
		return price;
	}
	
	public void setPrice(double price) {
		this.price = price;
	}
	
	public double getWeight() {
		return weight;
	}
	

	public void setWeight(double weight) {
		this.weight = weight;
	}

	public String getProducerName() {
		return producerName;
	}

	public void setProducerName(String producerName) {
		this.producerName = producerName;
	}
	
	public void Show() {
		System.out.println(price + " " + weight + " " + producerName);
	}
	
//	public ArrayList<CampingProduct> sort() {
//		return 
//	}
}

File: Lab2/Lab2/models/CampingProduct.py
from abc import ABC, abstractmethod

class CampingProduct(ABC):
    def __init__(self, producerName = "none", price = 0, weight = 0):
        self.producerName = producerName
        self.price = price
        self.weight = weight
    def get_producerName(self):
        return self.producerName
    def set_producerName(self, producerName):
        self.producerName = producerName
    def get_price(self):
        return self.price
    def set_price(self, price):
        self.price = price
    def get_weight(self):
        return self.weight
    def set_weight(self, weight):
        self.weight = weight
    @abstractmethod
    def Show(self):
        print("Виробник: " + self.producerName + "\nЦіна: " + str(self.price) + "\nВага: " + str(self.weight))